File: src/main/java/app/application/YoutubeDownloader.java
package app.application;

import app.MyApp;
import app.application.controller.main_window.MainWindowController;
import javafx.application.Application;
import javafx.application.Platform;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import net.rgielen.fxweaver.core.FxWeaver;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.context.ConfigurableApplicationContext;

public class YoutubeDownloader extends Application {

    private ConfigurableApplicationContext applicationContext;

    @Override
    public void init() throws Exception {
        this.applicationContext = new SpringApplicationBuilder().sources(MyApp.class)
                .run(getParameters().getRaw().toArray(new String[0]));
    }

    @Override
    public void start(Stage stage) throws Exception {
        FxWeaver fxWeaver = applicationContext.getBean(FxWeaver.class);
        Parent root = fxWeaver.loadView(MainWindowController.class);
        Scene scene = new Scene(root);
        stage.setTitle("Youtube Downloader");
        stage.setScene(scene);
        stage.setResizable(false);
        stage.show();
    }

    @Override
    public void stop() throws Exception {
        applicationContext.close();
        Platform.exit();
    }
}

File: src/main/kotlin/app/application/YoutubeDownloader.kt
package app.application

import app.MyApp
import app.application.controller.main_window.MainWindowController
import app.application.spring.service.GlobalObjectHandler
import javafx.application.Application
import javafx.application.Platform
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage
import net.rgielen.fxweaver.core.FxWeaver
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.ConfigurableApplicationContext
import kotlin.system.exitProcess

class YoutubeDownloader : Application() {
    private lateinit var applicationContext: ConfigurableApplicationContext
    @Throws(Exception::class)
    override fun init() {
        applicationContext = SpringApplicationBuilder().sources(MyApp::class.java)
                .run(*parameters.raw.toTypedArray())
    }

    @Throws(Exception::class)
    override fun start(stage: Stage) {
        setupHostServices()
        val fxWeaver = applicationContext.getBean(FxWeaver::class.java)
        val root = fxWeaver.loadView<MainWindowController, Parent>(MainWindowController::class.java)
        val scene = Scene(root)
        stage.title = "Youtube Downloader"
        stage.scene = scene
        stage.show()
    }

    @Throws(Exception::class)
    override fun stop() {
        applicationContext.close()
        Platform.exit()
        exitProcess(0)
    }

    private fun setupHostServices() {
        val globalObjectHandler = applicationContext.getBean(GlobalObjectHandler::class.java)
        globalObjectHandler.hostServices = hostServices
    }
}